pkg/streamingpromql/types/data.go

```diff
@@ -20,6 +20,19 @@ type SeriesMetadata struct {
 	Labels labels.Labels
 }
 
+type SeriesMetadataSlice []SeriesMetadata
+
+// Append appends this SeriesMetadataSlice with the provided otherSeriesMetadata.
+func (s SeriesMetadataSlice) Append(tracker *limiter.MemoryConsumptionTracker, otherSeriesMetadata ...SeriesMetadata) (SeriesMetadataSlice, error) {
```

Comment:
The use of the special `SeriesMetadataSlice` type here feels a bit unnatural to me.

What if the signature was something like this?

```go
func AppendSeriesMetadata(tracker *limiter.MemoryConsumptionTracker, base []SeriesMetadata, others ...SeriesMetadata) (SeriesMetadataSlice, error) {

```